include/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2023 Valve Corporation
+# Copyright (c) 2023 LunarG, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+target_include_directories(VulkanLayerUtils PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+
+target_compile_features(VulkanLayerUtils PUBLIC cxx_std_17)
```

Comment:
What this line does is require a `compiler mode is at least C++ 17.`

Which means both `VulkanLayerUtils` and clients that link against `VulkanLayerUtils` will require at least `C++17`

This is needed to address 2 issues:

1.) A compilation error

```
[ 50%] Building CXX object CMakeFiles/find_package_example.dir/main.cpp.o
In file included from /Users/runner/work/Vulkan-Utility-Libraries/Vulkan-Utility-Libraries/tests/find_package/main.cpp:1:
/Users/runner/work/Vulkan-Utility-Libraries/Vulkan-Utility-Libraries/install/include/vulkan/utility/vk_layer_settings.h:33:47: error: a space is required between consecutive right angle brackets (use '> >')
typedef std::vector<std::pair<std::string, int>> List;
```

By default the find_package project doesn't specify a C++ standard so it default to C++98, yet the interface of our library uses C++11 syntax.

2.) C++ ABI compatibility

https://stackoverflow.com/questions/67839008/please-explain-the-c-abi

Not sure how best to handle this TBH aside from not using the STL in the interface of the library.

Reply:
note that in the call we decided collectively that the interface should be C so I am going to change it.